linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/query.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Util for building regexes that map over specifically formatted metric names.
+ * `Query.clientQuery().build()` matches all possible client metrics.
+ *  Use defined helpers to make the query more specific.
+ */
+
+var Query = function() {
+  function escape(string) {
+    return string.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
+  }
+
+  function toR(arr) {
+    if (arr === "*" || !arr || arr.length === 0) {
+      return "(.*)";
+    } else {
+      return "(" + _.map(arr, escape).join("|") + ")";
+    }
+  }
+
+  var generalQuery = function() {
+    var query = {
+      routerLabels: [],
+      metricNames: []
+    };
+    query.allRouters = function() {
+      query.routerLabels = "*";
+      return query;
+    }
+    query.withRouter = function(router) {
+      query.routerLabels = [router];
+      return query;
+    }
```

Comment:
Would it make sense to append to the routerLabels array here, rather than replace?  I could imagine somebody calling:

    Query.clientQuery().withRouter("foo").withRouter("bar")

And expecting that to match both routers.